contracts/src/TeleporterTokenSource.sol

```diff
@@ -95,28 +98,31 @@ abstract contract TeleporterTokenSource is
             => mapping(address destinationBridgeAddress => uint256 balance)
     ) public bridgedBalances;
 
-    uint256 public constant MAX_TOKEN_MULTIPLIER = 1e18;
-
     /**
      * @notice Initializes this source token bridge instance to send
      * tokens to the specified destination chain and token bridge instance.
      */
     constructor(
         address teleporterRegistryAddress,
         address teleporterManager,
-        address tokenAddress_
+        address tokenAddress_,
+        uint8 tokenDecimals_
     ) TeleporterOwnerUpgradeable(teleporterRegistryAddress, teleporterManager) {
         blockchainID = IWarpMessenger(0x0200000000000000000000000000000000000005).getBlockchainID();
         require(tokenAddress_ != address(0), "TeleporterTokenSource: zero token address");
+        require(
```

Comment:
seems strange that we're using a constant from a separate library to validate the constructor of this contract, also why're we increasing the max decimals from 18 to 30 here

Reply:
oh this token dedcimals check is added, not sure about 30 decimals being the limit

Reply:
the constant is used by both `TeleporterTokenSource` and `TeleporterTokenDestination`, and they don't both inherit from a common base contract, so the `TokenScalingUtils` seemed like the best "common" place to put it, but I'm definitely open to other ideas.

Regarding the `30`, see this comment: https://github.com/ava-labs/teleporter-token-bridge/pull/142#discussion_r1611967570